packages/katello/katello/hostname-change.rb

```diff
@@ -551,7 +551,10 @@ def run
           STDOUT.puts result
           STDOUT.puts 'Restarting puppet services'
           run_cmd('/bin/systemctl try-restart puppet')
-          run_cmd('foreman-maintain service restart --only puppetserver')
+          `systemctl is-active puppetserver --quiet`
+          if $?.success?
+            run_cmd('systemctl try-restart puppetserver')
+          end
```

Comment:
Thoughts on parallel restarts?
```ruby
services = ['puppet']
`systemctl is-active puppetserver --quiet`
services << 'puppetserver' if $?.success?
run_cmd('/bin/systemctl try-restart ' + ' '.join(services))

Reply:
No strong feelings either way.

@ehelms?

Reply:
Sounds fine to me for puppet services. Now let me ask this question. In a world where Puppet is disabled, like Katello's default currently, would `puppet` be enabled? i.e. does this code need to also do the same sort of check on `puppet` ?

Reply:
No. The puppet service is always there (as the installer is using Puppet Agent), but a "try-restart" will only restart it if it's also *running*.

Reply:
well, look at this